pkg/cmd/update-tls-artifacts/generate-owners/generate_owners_options.go

```diff
@@ -28,51 +22,19 @@ func (o *GenerateOwnersOptions) Run() error {
 	if err != nil {
 		return err
 	}
-	ownershipJSONBytes, err := json.MarshalIndent(result, "", "    ")
-	if err != nil {
-		return err
-	}
-	markdown, err := GenerateOwnershipMarkdown(result)
-	if err != nil {
-		return err
-	}
-	violations := GenerateViolationJSON(result)
-	violationJSONBytes, err := json.MarshalIndent(violations, "", "    ")
+	violations, err := metadata.GenerateViolationList(result, metadata.All...)
```

Comment:
your division here I think is close. You want something like

```go

type CertificateFeature interface{
  Get<Foo>JSON([]certgraphapi.PKIList) (bytes, error)
  Get<Foo>ViolationJSON([]certgraphapi.PKIList) (bytes, error)
  Get<Foo>Markdown([]certgraphapi.PKIList) (bytes, error)
  Get<Foo>ViolationMarkdown([]certgraphapi.PKIList) (bytes, error)
}

```

And then wrap that.  I think this is close, but we want to make individuals completely responsible for what they add.

Reply:
I've realized this file now generates for all violations. Do we want to have separate subcommands for owners/description (and make people copy it whenever a new one is added)?
iiuc there's no harm in generating all files now